go/ioutil2/ioutil_test.go

```diff
@@ -0,0 +1,17 @@
+package ioutil2
+
+import (
+	"os"
+	"testing"
+)
+
+func TestWrite(t *testing.T) {
+	fname := "/tmp/atomic-file-test.txt"
+	err := WriteFileAtomic(fname, []byte("test string\n"), 0664)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Remove(fname); err != nil {
```

Comment:
Before removing, maybe the test could read back the file and check that all the data is there.